### ISSUE TITLE
Support Workload Identity

### ISSUE THREAD

**Issue by tam7t** [GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp#11]
### TL;DR

The CSI driver should use the identity of the pod when fetching secrets instead of the instance identity.

### Design

**Proposal**

Modify the plugin to fetch credentials that are compatible with [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) by default to promote least privilege permissions on secrets.

Grant the CSI driver the `serviceaccounts/token` permission in the k8s cluster and use the `pod.uid` to fetch tokens for the pod to then communicate with the `secretmanager.googleapis.com` API.

**Alternatives considered**

**Resources**

**Additional information**
